### DISCUSSION TITLE
Can I make the bar or margins disappear the moment I switch to max layout without restarting qTile?

### DISCUSSION THREAD

**Discussion by prateekamana** [qtile/qtile#3617] [Q&A]
I figured it should have something to do with hooks, but I don't know what to do in hooks, I could change the script itself but I realized that for those changes to reflect I'll have to restart qTile.

**Comment by ervinpopescu** [comment]
I think you can use ***subscribe.layout_change*** for that. More details here: http://docs.qtile.org/en/latest/manual/ref/hooks.html#libqtile.hook.subscribe.layout_change

I'm not at home right now, but I'll try to make it happen when I am.

**Reply by prateekamana** [reply]
Yes but I can't just do something like
```
if(non_max_layout){
top=bar.Bar()
}
```
right? Because if I make changes to the config.py then I will have to restart qtile. Is there a method in qTile to make the bar disappear?

**Reply by elParaguayo** [reply]
What's wrong with restarting qtile?

There's also lazy.reload_config() for just reloading the config.

**Reply by ervinpopescu** [reply]
> Yes but I can't just do something like
> ```
> if(non_max_layout){
> top=bar.Bar()
> }
> ```

I don't get why you used non-python code but ok. xD

> right? Because if I make changes to the config.py then I will have to restart qtile.

Yes, you are right.

> Is there a method in qTile to make the bar disappear?

That would be: 

```
qtile.cmd_hide_show_bar()
```

I didn't have time for this, I'll try to find some time in the next 3 hours or so.

**Reply by ervinpopescu** [reply]
Do you want the max layout to not have a margin?
If yes, you can just set `margin=0` in the `layouts` variable... 

```py
@hook.subscribe.layout_change
def on_max_layout(layout, group):
    if layout.name == "max":
        layout.margin = 0  # don't do this 
        group.qtile.cmd_hide_show_bar()
```

**Reply by prateekamana** [reply]
I am trying this but since this is a toggle function, this also ends up showing showing the bar if it's already hidden once I switch to max layout.

**Reply by prateekamana** [reply]
Also, I have no margins on the max layout, it's already 0, but the margins are coming from the bar margins that I set up to make the windows equidistant 
![image](https://user-images.githubusercontent.com/41776584/172665441-42d37c80-6273-4501-9114-5159a02c1773.png)

**Reply by mgutz** [reply]
> I am trying this but since this is a toggle function, this also ends up showing showing the bar if it's already hidden once I switch to max layout.

I did something similar. Since there is not an explicit flag to hide, track the state.

```python
last_layout = ""

@hook.subscribe.layout_change
def layout_change(layout, _):
    global last_layout

    if layout.name == "max":
        qtile.cmd_hide_show_bar()
        last_layout = "max"
    elif last_layout == "max":
        qtile.cmd_hide_show_bar()
        last_layout = ""
```

**Reply by prateekamana** [reply]
That's a cool solution and I don't even get it immediately because I am sleep deprived. Thank you, can you please tell me how I can reset the bar margins and the bar.Gap() margins to 0 now?
Edit: Your solution didn't work so I ended up doing this 
```python
is_bar_hidden = False
@hook.subscribe.layout_change
def layout_changed(layout, group):
    global is_bar_hidden
    if (not is_bar_hidden) and layout.name == 'max':
        qtile.cmd_hide_show_bar('top')
        is_bar_hidden = True
    elif is_bar_hidden and layout.name != 'max':
        qtile.cmd_hide_show_bar('top')
        is_bar_hidden = False
```
